### PR TITLE
importinto: fix incorrect index kv group count when table have common handle

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #50752

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/disttask/importinto/encode_and_sort_operator_test.go [hunk 1]
```diff
@@ -185,6 +185,12 @@ func TestGetWriterMemorySizeLimit(t *testing.T) {
...
 			perIndexKVMemSizePerCon: 153391689,
 		},
+		{
+			createSQL:               "create table t (a int, b int, c int, primary key(a,b,c) clustered, key(b,c), unique(b), unique(c), key(a,b))",
+			numOfIndexGenKV:         4,
+			dataKVMemSizePerCon:     460175067,
+			perIndexKVMemSizePerCon: 153391689,
+		},
 		{
 			createSQL:               "create table t (a int, b int, c int, primary key(a,b,c) nonclustered, key(b,c), unique(b), unique(c), key(a,b))",
```
Comment (lance6716) [inline pkg/disttask/importinto/encode_and_sort_operator_test.go:195]: Do it cover the `ClusteredIndexDefModeIntOnly`? check with that option and a table with non-int PK

Comment (D3Hunter) [reply pkg/disttask/importinto/encode_and_sort_operator_test.go:195]: > // ClusteredIndexDefModeIntOnly indicates only single int primary key will default be clustered.

seems not related, composite int pk is also CommonHandle

Comment (lance6716) [reply pkg/disttask/importinto/encode_and_sort_operator_test.go:195]: OK I see that `HasClusteredIndex` can both check `t.PKIsHandle || t.IsCommonHandle`

**Review by lance6716 — APPROVED** [review]

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Pull request merged**